diff --git a/dlls/windowscodecs/windowscodecs.spec b/dlls/windowscodecs/windowscodecs.spec
index ae3660c6f3b..cfaff9703ac 100644
--- a/dlls/windowscodecs/windowscodecs.spec
+++ b/dlls/windowscodecs/windowscodecs.spec
@@ -114,5 +114,5 @@
 @ stdcall WICMapSchemaToName(ptr wstr long ptr ptr)
 @ stdcall WICMapShortNameToGuid(wstr ptr)
 @ stub WICMatchMetadataContent
-@ stub WICSerializeMetadataContent
+@ stdcall WICSerializeMetadataContent(ptr ptr long ptr)
 @ stdcall WICSetEncoderFormat_Proxy(ptr ptr ptr ptr)
