diff --git a/dlls/coremessaging/private.h b/dlls/coremessaging/private.h
index bf1a2fca7a1..b423897b00b 100644
--- a/dlls/coremessaging/private.h
+++ b/dlls/coremessaging/private.h
@@ -40,6 +40,8 @@
 
 #include "async_private.h"
 
+extern IActivationFactory *dispatcher_queue_activation_factory;
+
 HRESULT async_action_create( IUnknown *invoker, async_operation_callback callback, IAsyncAction **out );
 
 #define DEFINE_IINSPECTABLE_( pfx, iface_type, impl_type, impl_from, iface_mem, expr )             \
